docs/upgrades.md

```diff
@@ -18,6 +18,11 @@ versions. Here are all version vars for each component:
 * flannel_version
 * kubedns_version
 
+Please remember that some components upgrade depends on minimum installed version.
+In example **calico version 2.6.5 upgrade to 3.1.1 is upgrading etcd store to etcdv3**.
+The kubespray stack upgrade would failed when calico version is below 2.6.5. Please check
```

Comment:
Do we have a preflight check for this so it does not happen? Just updating the docs seems insufficient.

Reply:
@woopstar I run the whole cluster upgrade and the problem was that calico version checks was run during calico installation not on the beginning of run. If you have calico i.e. 2.6.2, upgrade of cluster begin, pass about 8 spets of upgrade and failed on: https://github.com/kubernetes-incubator/kubespray/blob/7f91f6e03409016f2cce3de89f504ad186d582a8/upgrade-cluster.yml#L85
Then you need to pull old calico role for i.e. 2.6.8 upgrade calico stack and rerun upgrade-cluster.yaml.

Reply:
Wouldn't it be smart to add a pre-check that checks your calico version and if it's older than 2.6.5 you cannot upgrade to >3.x ?

Reply:
Seems ok for me I can add it. What about other components should we add some global checks for other k8s stack components?

Reply:
@woopstar as you mentioned I add calico precheck before upgrade. A little not sure if it is good place or we should different task for all cluster components version checks if more version dependencies exist?